### ISSUE TITLE
Deploy Chalice app via code vs. command line

### ISSUE THREAD

**Issue by melvinev** [aws/chalice#1582]
hi great package -

Is it possible to deploy Chalice app via a method of the Chalice() class?

Something like this: 
app = Chalice()
app.deploy()

Instead of "chalice deploy" in terminal?

**Comment by jamesls** [comment]
No, there's no public API for doing this as of now, primarily due to no one asking for it.  Can you tell me more about how you would use this API if it existed?

**Comment by melvinev** [comment]
Hi @jamesls sorry I missed this.

Similar to Prefect - they have a CLI like "prefect register" but I can also call from within my Jupyter Notebook via flow.register().  The Python API is very handy.  It's just easier than having to go to a terminal to deploy and some of my CI/CD scripts would be simpler if I just was able to incorporate the re-deployment via Python instead of a terminal.
